lib/debug/variable_inspector.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require_relative 'limited_pp'
+
+module DEBUGGER__
+  class VariableInspector
+    def indexed_members_of(obj, start:, count:)
+      return [] if start > (obj.length - 1)
```

Comment:
In this implementation, the `#class` will disappear when displaying Array. Is this intentional behavior?

Reply:
That makes sense (in the same spirit as https://github.com/ruby/debug/pull/1004), though perhaps we should keep the `#class` if it's a non-standard subclass of `Array`. I'll make that in a separate PR.